pkg/sidecarlogresults/sidecarlogresults.go

```diff
@@ -0,0 +1,190 @@
+	sleepTime := time.Millisecond
+	for true {
+		f, err := os.Open(resultsDir)
+		if err != nil {
+			return err
+		}
+		defer f.Close()
+		files, err := f.Readdir(0)
+		if err != nil {
+			return err
+		}
+		for _, v := range files {
+			if _, ok := resultParsed[v.Name()]; !ok {
+				resultParsed[v.Name()] = false
+			}
+			if expectedResults.Has(v.Name()) && !resultsFetched.Has(v.Name()) {
+				value, err := os.ReadFile(fmt.Sprintf("%v/%v", resultsDir, v.Name()))
+				if err != nil {
+					return err
+				}
+				newResult := SidecarLogResult{
+					Name:  v.Name(),
+					Value: string(value),
+				}
+				if val, ok := results[newResult.Name]; ok {
+					if val == newResult.Value {
+						if newResult.Value == "" {
+							now := time.Now().UnixMilli()
+							if _, ok := startTimesPerResult[newResult.Name]; !ok {
+								startTimesPerResult[newResult.Name] = now
+							} else if now-startTimesPerResult[newResult.Name] >= millisecondsTowaitForEmptyResult {
+								// value is still empty after second from the time that the result file was created.
+								resultParsed[newResult.Name] = true
+							}
+						} else {
+							// populated value is not changing anymore after 50 ms.
+							// assume that the contents have been fully written.
+							resultParsed[newResult.Name] = true
+						}
+					}
+				}
+				results[newResult.Name] = newResult.Value
+
+				if resultParsed[newResult.Name] == true {
+					newResult.log()
+					resultsFetched.Insert(v.Name())
+				}
+			}
+		}
+		f.Close()
+		time.Sleep(sleepTime)
+		if reflect.DeepEqual(resultsFetched, expectedResults) {
+			break
+		}
+	}
+	return nil
+}
+
+// GetResultsFromSidecarLogs extracts results from the logs of the results sidecar
+func GetResultsFromSidecarLogs(ctx context.Context, clientset kubernetes.Interface, namespace string, name string, container string, sidecarLogResults []v1beta1.PipelineResourceResult) ([]v1beta1.PipelineResourceResult, error) {
+	podLogOpts := corev1.PodLogOptions{Container: container}
+	req := clientset.CoreV1().Pods(namespace).GetLogs(name, &podLogOpts)
+	sidecarLogs, err := req.Stream(ctx)
+	if err != nil {
+		podInitializingError := fmt.Sprintf(`container "%v" in pod "%v" is waiting to start: PodInitializing`, container, name)
+		if err.Error() == podInitializingError {
+			return sidecarLogResults, nil
+		}
+		return sidecarLogResults, err
+	}
+	defer sidecarLogs.Close()
+	maxResultLimit := config.FromContextOrDefaults(ctx).FeatureFlags.MaxResultSize
+	sidecarLogResults, err = extractResultsFromLogs(sidecarLogs, sidecarLogResults, maxResultLimit)
+	return sidecarLogResults, err
+}
+
+func extractResultsFromLogs(logs io.Reader, sidecarLogResults []v1beta1.PipelineResourceResult, maxResultLimit int) ([]v1beta1.PipelineResourceResult, error) {
+	scanner := bufio.NewScanner(logs)
+	buf := make([]byte, 2)
+	scanner.Buffer(buf, 2*maxResultLimit)
+	resStr := ""
+	for scanner.Scan() {
+		text := scanner.Text()
+		if text != resultFinisher {
```

Comment:
This delimiter style encoding is brittle. Would suggest using a json encoded string starting and ending with `"` or `'` instead. In the future if/when we support array and dictionary results this will gives us an easy upgrade path approach to follow as a benefit.

Reply:
Sure! I can do that. 
I did want to point out that this mechanism already supports array and object results. Both arrays and object results are written as strings in the results's file.

Reply:
done!